### PR TITLE
Loader: Remove Handlers.

### LINKED ISSUE #15953: Suggestion: Clearer Loader.Handlers API
##### Description of the problem

`Loader.Handlers` is used in four high-level asset loaders (glTF, FBX, MTL, MMD) under `examples/js/loaders` for providing an API that allows the use of custom texture (or other type of low-level assets) loader in them. For example, users can override regular `TextureLoader` with custom texture loader supporting `ImageBitmap` and/or compressed texture.

Example code.

```javascript
function CustomTextureLoader() {
}

CustomTextureLoader.prototype.load = function ( url, onLoad, onProgress, onError ) {
    ...
};

THREE.Loader.Handlers.add( /\.(png|jpg|jpeg)$/, new CustomTextureLoader() );

var loader = THREE.GLTFLoader();
loader.load( ... ); // CustomTextureLoader used inside instead of regular TextureLoader
```

I think current `Loader.Handlers` API has a problem. It's very unclear for users how and where added handler is used because no clear documents and `THREE.Loader.Handlers` is global scope. Even if we document, users still need to read through the document and carefully add handler to let only the target high-level asset loader use the custom low-level loader inside. Unless carefully doing that, the custom low-level loader can be unexpectedly used by other high-level asset loaders because `Loader.Handler` is global scope.

I think loader scope `Handlers` is better than global scope. Easier to understand and control for users.

Two ideas in my mind.

1. Move handlers to `LoadingManager` method. https://github.com/mrdoob/three.js/issues/15935#issuecomment-471050306

Pros: No need to add new public method to any high-level loaders.
Cons: Users need to read loader's document to know how added handler is used inside.

```javascript
// user code
var manager = new THREE.LoadingManager();
manager.addHandler( /\.(png|jpg|jpeg)$/, new CustomTextureLoader() );
var loader = new THREE.FooLoader( manager );
loader.load( ... );

// in FooLoader
FooLoader.prototype.varFunction = function () {
    var textureLoader = this.manager.getHandler( imageUrl );
    if ( textureLoader === null ) textureLoader = new THREE.TextureLoader( this.manager );
    textureLoader.load( ... );
};
```

2. Each asset loader explicitly has overriding loader method with clear name. https://github.com/mrdoob/three.js/pull/15919#issuecomment-470760573

Pros: Easier to understand for users.
Cons: We need to add new public method to each high-level asset loader.

```javascript
// user code
var loader = new THREE.FooLoader();
loader.setCustomTextureLoader( new CustomTextureLoader() );

// in FooLoader
FooLoader.prototype.setCustomTextureLoader = function ( loader ) {
    this._textureLoader = loader;
};

FooLoader.prototype.varFunction = function () {
    var textureLoader = this._textureLoader || new THREE.TextureLoader( this.manager );
    textureLoader.load( ... );
}
```

I prefer 2. Any thoughts?

My personal short term goal is having clear API which enables `ImageBitmap` in `GLTFLoader`.

##### Three.js version

- [x] Dev
- [ ] r102
- [ ] ...

##### Browser

- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] Android
- [ ] iOS

### PR DESCRIPTION
**Pull request by Mugen87**
Fixed #15953

This PR removes `Loader.Handler` and enhances the API of `LoadingManager` instead. The following methods are introduced:

- `.addHandler()`
- `.removeHandler()`
- `.getHandler()`

### CODE REVIEW

File: examples/js/loaders/deprecated/LegacyGLTFLoader.js [hunk 1]
```diff
@@ -1079,7 +1079,7 @@ THREE.LegacyGLTFLoader = ( function () {
 
 						}
 
-						var textureLoader = THREE.Loader.Handlers.get( sourceUri );
+						var textureLoader = options.manager.getHanlder( sourceUri );
```
Comment (mrdoob) [inline examples/js/loaders/deprecated/LegacyGLTFLoader.js]: Typo: `getHanlder`

Comment (Mugen87) [reply examples/js/loaders/deprecated/LegacyGLTFLoader.js]: Fixed 👍

**Comment by mrdoob** [comment]
Thanks!

**Comment by Mugen87** [comment]
`THREE.Loader` is so nice now 🎉😊

Updated migration guide: https://github.com/mrdoob/three.js/wiki/Migration-Guide#r108--r109

**Comment by mrdoob** [comment]
> `THREE.Loader` is so nice now 🎉😊

Indeed! As it should have been from the start 😅

**Comment by mrdoob** [comment]
I think it can be nicer actually:

```js
function Loader( manager ) {

	this.manager = ( manager !== undefined ) ? manager : DefaultLoadingManager;

}

Object.assign( Loader.prototype, {

	crossOrigin: 'anonymous',
	path: '',
	resourcePath: '',

	load: function ( /* url, onLoad, onProgress, onError */ ) {},

	parse: function ( /* data */ ) {},

	setCrossOrigin: function ( crossOrigin ) {

		this.crossOrigin = crossOrigin;
		return this;

	},

	setPath: function ( path ) {

		this.path = path;
		return this;

	},

	setResourcePath: function ( resourcePath ) {

		this.resourcePath = resourcePath;
		return this;

	}

} );

export { Loader };
```

**Comment by Mugen87** [comment]
What is the difference between assigning the properties to `prototype` and `this`? When converting to classes, it should be actually the latter one anyway.

**Comment by mrdoob** [comment]
If I remember correctly... by defining them in the prototype, you don't create new objects every time you call the constructor, it reuses the ones in the prototype, but if you call loader.setPath() it redefines `this.path` and doesn't use the one in the constructor anymore.

**Pull request merged**